tests/storage/migration.go

```diff
@@ -401,6 +401,71 @@ var _ = Describe(SIG("Volumes update with migration", decorators.RequiresTwoSche
 			waitForMigrationToSucceed(virtClient, vm.Name, ns)
 		})
 
+		It("should trigger the migration once the destination DV exists", func() {
+			volName := "disk0"
+			srcDV := createDV()
+			vm := createVMWithDV(srcDV, volName)
+			destDV := libdv.NewDataVolume(
+				libdv.WithBlankImageSource(),
+				libdv.WithStorage(libdv.StorageWithStorageClass(testSc),
+					libdv.StorageWithVolumeSize(size),
+					libdv.StorageWithVolumeMode(k8sv1.PersistentVolumeFilesystem),
+					libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce),
+				),
+			)
+			By("Update volumes")
+			i := slices.IndexFunc(vm.Spec.Template.Spec.Volumes, func(volume virtv1.Volume) bool {
+				return volume.Name == volName
+			})
+			Expect(i).To(BeNumerically(">", -1))
+			By(fmt.Sprintf("Replacing volume %s with DV %s", volName, destDV.Name))
+
+			updatedVolume := virtv1.Volume{
+				Name: volName,
+				VolumeSource: virtv1.VolumeSource{DataVolume: &virtv1.DataVolumeSource{
+					Name: destDV.Name,
+				}}}
+
+			p, err := patch.New(
+				patch.WithRemove("/spec/dataVolumeTemplates"),
+				patch.WithReplace(fmt.Sprintf("/spec/template/spec/volumes/%d", i), updatedVolume),
+				patch.WithReplace("/spec/updateVolumesStrategy", virtv1.UpdateVolumesStrategyMigration),
+			).GeneratePayload()
+			Expect(err).ToNot(HaveOccurred())
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, p, metav1.PatchOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() []virtv1.VirtualMachineInstanceCondition {
+				vmi, err := virtClient.VirtualMachineInstance(ns).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vmi.Status.Conditions
+			}, 30*time.Second, time.Second).Should(ContainElement(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+				"Type":    Equal(virtv1.VirtualMachineInstanceVolumesChange),
+				"Status":  Equal(k8sv1.ConditionFalse),
+				"Message": ContainSubstring("One of the destination volumes doesn't exist"),
+			})))
+
+			By("Create the destination DV")
+			_, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(ns).Create(context.Background(),
+				destDV, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			waitForMigrationToSucceed(virtClient, vm.Name, ns)
+		})
+
+		It("should trigger the migration once the destination PVC exists", func() {
+			volName := "disk0"
+			srcDV := createDV()
+			vm := createVMWithDV(srcDV, volName)
+			By("Update volumes")
+			updateVMWithPVC(vm, volName, destPVC)
+
```

Comment:
dont you want to have the same check here too?:
			Eventually(func() []virtv1.VirtualMachineInstanceCondition {
				vmi, err := virtClient.VirtualMachineInstance(ns).Get(context.Background(), vm.Name, metav1.GetOptions{})
				Expect(err).ToNot(HaveOccurred())
				return vmi.Status.Conditions
			}, 30*time.Second, time.Second).Should(ContainElement(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
				"Type":    Equal(virtv1.VirtualMachineInstanceVolumesChange),
				"Status":  Equal(k8sv1.ConditionFalse),
				"Message": ContainSubstring("One of the destination volumes doesn't exist"),
			})))

Reply:
we don't care too much imo :D the important piece is that the migration succeeds after its creation

Reply:
yeah I understand that we dont really care just to make sure that the creation of the pvc doesnt happen too fast such that the migration doesnt even identify that the pvc didnt exist

Reply:
ok I see let me add a check so we can slow down the test

Reply:
Added a check that the migration is pending. It will be in pending in any case, but at least we can guarantee that the PVC creation happens after the VMIM object creation